tensorboard/writer/event_file_writer.py

```diff
@@ -0,0 +1,229 @@
+# Copyright 2015 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Writes events to disk in a logdir."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os.path
+import socket
+import threading
+import time
+
+import six
+
+from tensorboard.compat.proto import event_pb2
+from tensorboard.writer.record_writer import RecordWriter, directory_check
+
+
+class EventsWriter(object):
+    '''Writes `Event` protocol buffers to an event file.'''
+
+    def __init__(self, file_prefix, filename_suffix=''):
+        '''
+        Events files have a name of the form
+        '/some/file/path/[file_prefix].out.tfevents.[timestamp].[hostname]'
+
+        Args:
+          file_prefix: The string that will be prepended to
+            the filename of the event file.
+          filename_suffix: The string that will be appended to
+            the filename of the event file.
+        '''
+        self._file_name = file_prefix + ".out.tfevents." + str(time.time())[:10] + "." +\
+            socket.gethostname() + filename_suffix
+        self._num_outstanding_events = 0
+        self._py_recordio_writer = RecordWriter(self._file_name)
+        # Initialize an event instance.
+        self._event = event_pb2.Event()
```

Comment:
We should also set the `file_version` field to `"brain.Event:2"` to match this line:
https://github.com/tensorflow/tensorflow/blob/3109adb52056659e1756002771e3593975804f15/tensorflow/core/util/events_writer.cc#L91

Reply:
Yup, write this information by event_file_writer makes more sense. (I was put it in the pytorch side)